pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/FunctionsBase.java

```diff
@@ -692,4 +693,25 @@ public StreamingOutput downloadFunction(
     public List<ConnectorDefinition> getConnectorsList() throws IOException {
         return functions.getListOfConnectors();
     }
+
+    @PUT
+    @ApiOperation(value = "Updates a Pulsar Function on the worker leader")
+    @ApiResponses(value = {
+            @ApiResponse(code = 403, message = "The requester doesn't have super-user permissions"),
+            @ApiResponse(code = 404, message = "The function does not exist"),
+            @ApiResponse(code = 400, message = "Invalid request"),
+            @ApiResponse(code = 307, message = "Redirecting to the worker leader"),
+            @ApiResponse(code = 200, message = "Pulsar Function successfully updated")
+    })
+    @Path("/leader/{tenant}/{namespace}/{functionName}")
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    public void updateFunctionOnWorkerLeader(final @PathParam("tenant") String tenant,
```

Comment:
I don't think it is a good idea to expose this in the admin API.  This is an internal thing and users should not have to the option to call this directly